doc/LibcurlTransportAdapter.md

```diff
@@ -0,0 +1,162 @@
+
+### Poor man's substitute
+
+As mentioned before, using libcurl properly means delegating TCP and HTTP entirely to libcurl. In a nutshell, libcurl works as a state machine. Calling [curl_easy_perform(handle)](https://curl.se/libcurl/c/curl_easy_perform.html) is the way to start the machine (step 4 mentioned above). A loop is started and won't be completed until the request (set it up in the handle) is completed. If the request represents a GET operation to download data, and the application is expecting to read the data coming for the server, the handle needs to be configured with a _delegated function_ (a callback) that libcurl will invoke as soon as it has data from the server. Every time the callback is invoked, libcurl will inform how big it is the chunk of data that has arrived from the server, and a pointer to the buffer where data can be read. The next flow would be an approximation summary of what is happening within libcurl while dispatching a GET request:
+ 
+1.	Resolve DNS from request URL.
+2.	Get a network socket from the OS.
+3.	Establish TCP connection to the server.
+4.	Secure connection with TLS when using HTTPS.
+5.	Create an HTTP request from the handle and write it to the network socket.
+6.	Wait for socket to be readable (means server has responded) and get an HTTP status line and headers.
+7. If there's a payload body (data to be downloaded), a loop is started here as:
+*  7.1 Read data from the network socket.
+*  7.2 Invoke the application callback passing the data read from the socket.
+* 7.3 Repeat until the entire response has been received.
+ 
+This flow demonstrates how using libcurl to download data requires developers to use a callback pattern. For the example of reproducing a media file (mentioned above), playing media must happen as part of the callback that is invoked by libcurl, which is very unfortunate. 
+ 
+There are valid alternatives for applications, but it might be complex approaches. For example, using a multi-threading strategy, an application can call curl_easy_perform from one thread and use the read callback to copy the data from the response into a memory buffer. Then, another thread can play the media from the buffer. This strategy requires threads synchronization and complicates the debugging experience.
+ 
+Some other designs might help applications to deal with this. But it is a different story for libraries.  The Azure SDK is a library, and it ensures a consistent experience regardless of the programing language, as such, the API is not shaped by its implementation details. The libcurl implementation details from the Azure SDK is considered a poor man's substitute by [libcurl author](https://github.com/curl/curl/issues/6009#issuecomment-698611402), and it is explained next.
```

Comment:
I'm not sure this paragraph is necessary.

Reply:
It is the explanation of why the approach is called `a poor man substitute`.  As that's the name given by libcurl author.
But, yes, I think I know what you mean.  I was trying to add some fun/drama around it;  but it is not adding any real technical value >D

Reply:
Drama is wonderful when telling stories (arguably it's essential). But it's not necessarily a great solution in a technical document :(.

FWIW, the phrase "Poor Man's" does not appear in the policheck database, nor did any online definitions I found indicate it was considered inappropriate. But it is a term I wouldn't necessarily use.